File: raspberry/SmartHome/app/src/main/java/smarthome/raspberry/server/httphandlers/ErrorHandler.java
package smarthome.raspberry.server.httphandlers;

import fi.iki.elonen.NanoHTTPD;

public class ErrorHandler extends BaseRequestHandler{

    public NanoHTTPD.Response getError(String message) {
        return getInvalidRequestResponse(message);
    }

    @Override
    public NanoHTTPD.Response serve(NanoHTTPD.IHTTPSession session) {
        throw new IllegalArgumentException(ErrorHandler.class.getSimpleName() +
                " can't serve for valid session");
    }
}

File: raspberry/SmartHome/app/src/main/java/smarthome/raspberry/server/httphandlers/ErrorHandler.kt
package smarthome.raspberry.server.httphandlers

import fi.iki.elonen.NanoHTTPD

class ErrorHandler : BaseRequestHandler() {

    fun getError(message: String): NanoHTTPD.Response {
        return getInvalidRequestResponse(message)
    }

    override suspend fun serve(session: NanoHTTPD.IHTTPSession): NanoHTTPD.Response {
        throw IllegalArgumentException(ErrorHandler::class.java.simpleName + " can't serve for valid session")
    }
}
